### DISCUSSION TITLE
Syslog-ng client certificate validity check at software startup

### DISCUSSION THREAD

**Discussion by BRKiranAT** [syslog-ng/syslog-ng#5200] [Q&A]
We have configured syslog-ng with mutual TLS authentication, and we want syslog-ng client to be capable of managing its (client) certificate also. As the certificate is a configuration element of syslog-ng, it needs to be verified during software startup, like all other configuration elements. 
syslog-ng client must validate the following properties of its certificate: 
Availability => if the client certificate is present in the certificate folder (/etc/pki/tls/certs/).
Signature => The presence of a valid CA
Date => the validity beginning and expiry of certificate
Key size => 2048 for RSA & 256 for EC certificates
SAN field => the SAN must match the IP of the server
Extended key usage => must be set to client authentication
If any of these are found to be invalid, syslog-ng client must write appropriate error message for rejecting the certificate in the log file.

I appreciate your attention to this and look forward to your support in implementing this enhancement.

**Comment by HofiOne** [comment]
Did you follow the instructions provided here to set up the mutual authentication for both sides?

https://syslog-ng.github.io/admin-guide/100_TLS-encrypted_message_transfer/002_Mutual_authentication_using_TLS/README

I cannot re-check it now, but as I can remember you will get proper error messages on both sides if the authentication fails, and about why it failed.

If you think this is not the case, please provide 
- excat problem you have discovered
- exact steps to reproduce the issue
thanks!

**Reply by BRKiranAT** [reply]
There are no issues with mutual authentication, it is functioning properly. Our requirement is that while starting syslog ng as a client, the syslog-ng client should verify its own certificate.  how to do it?

**Reply by HofiOne** [reply]
I will double-check which fields are validated, but I'm sure syslog-ng makes a basic validation of the certificates, though, as I can recall surely not all of them.

**Reply by BRKiranAT** [reply]
Please let us know which fields are validated. it is blocking for delivery.
